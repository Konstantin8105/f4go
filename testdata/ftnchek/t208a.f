      SUBROUTINE  DDIAPA (IX,IY)
C     (Add to Path)
C     Add the point (IX,IY) to the current path, which is assumed
C     to have  already been  started by  a call  to DDIBPA.   The
C     current point and visibility are updated in COMMON.
C     (27-May-1991)
C- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
C
C     EXTERNAL REFERENCES (FUNCTION,SUBROUTINE,COMMON)
C
C     EXTERNAL REFS       TKIOLL,      TKIOWB,      TKIOWH,      TKIOWN
C
      INTEGER             TKIOLL
C- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
C
C     NON-COMMON VARIABLES
C
C     SETS                BLANK
C
      INTEGER             BLANK,       DUMMY,       IX,          IY
C
C-----------------------------------------------------------------------
C                         P o s t S c r i p t
C           D i s p l a y   D e v i c e   I n t e r f a c e
C                       C O M M O N   B l o c k
C
C     DSSIZE                   display surface size in cm
C     IMAGE                    image transformation in effect
C     (LASTX,LASTY,VILAST)     last position and visibility of pen
C     LINEIN                   line intensity (0..1 scale)
C     LINEWT                   line weight (1..25 scale)
C     MAGFAC                   resolution magnification factor
C     (MOVEX,MOVEY)            position of first point in path
C     NPATH                    number of line segments in current page
C     PLSTEP                   plotter step size in cm
C     (PSXOFF,PSYOFF)          image offset in PostScript units (bp)
C     ROTATE                   plot frame is rotated
C     (SX,SY)                  scale factors (unit square to plot steps)
C     TIMAGE(*,*)              image transformation matrix
C     (XMAX,YMAX,ZMAX)         normalized device space extents
C     (MAXX,MAXY,MINX,MINY)    actual coordinate limits reached
C
C-----------------------------------------------------------------------
      INTEGER             LASTX,       LASTY,       LINEWT,      MAGFAC
      INTEGER             MAXX,        MAXY,        MINX,        MINY
      INTEGER             MOVEX,       MOVEY,       NPATH,       PSXOFF
      INTEGER             PSYOFF
C
      LOGICAL             DVINIT,      IMAGE,       ROTATE,      VILAST
C
      REAL                DSSIZE,      LINEIN,      PLSTEP,      SX
      REAL                SY,          TIMAGE,      XMAX,        YMAX
      REAL                ZMAX
C
      COMMON / DDI01  /   DSSIZE,      LINEIN,      PLSTEP,      SX
      COMMON / DDI01  /   SY,          TIMAGE(4,4), XMAX,        YMAX
      COMMON / DDI01  /   ZMAX,        LASTX,       LASTY,       LINEWT
      COMMON / DDI01  /   MAGFAC,      MAXX,        MAXY,        MINX
      COMMON / DDI01  /   MINY,        MOVEX,       MOVEY,       NPATH
      COMMON / DDI01  /   PSXOFF,      PSYOFF,      DVINIT,      IMAGE
      COMMON / DDI01  /   ROTATE,      VILAST
C
C
      DATA                BLANK     / 32 /
C
C     We can generate an  entry of size "-ddddd -ddddd R "; try to
C     keep line lengths under 80 characters.
C
      IF (TKIOLL(DUMMY) .GT. 64) CALL TKIOWH (2H$N,2)
C
C     Use relative coordinates for lines to reduce the  number of
C     bytes in the  verbose  PostScript  command language.   This
C     shortens the plot file by 5 to 10 percent.
C
C     We optimize output into 3 different forms:
C
C     # # R   (relative lineto)
C     # X     (relative lineto with delta-y = 0)
C     # Y     (relative lineto with delta-x = 0)
C
C     The  latter two cases  occur frequently enough  to be worth
C     taking advantage of  to compress the  output and reduce the
C     parsing  time.    We cannot  discard zero length  segments,
C     however, because that  would discard single  points from  a
C     point plot.
C
      IF (IX .EQ. LASTX) GO TO 10
      IF (IY .EQ. LASTY) GO TO 20
      CALL TKIOWN (IX-LASTX)
      CALL TKIOWB (BLANK)
      CALL TKIOWN (IY-LASTY)
      CALL TKIOWH (3H R ,3)
      GO TO 30
   10 CALL TKIOWN (IY-LASTY)
      CALL TKIOWH (3H Y ,3)
      GO TO 30
   20 CALL TKIOWN (IX-LASTX)
      CALL TKIOWH (3H X ,3)
   30 NPATH = NPATH + 1
      LASTX = IX
      LASTY = IY
      VILAST = .TRUE.
      MAXX = MAX0(MAXX,IX)
      MINX = MIN0(MINX,IX)
      MAXY = MAX0(MAXY,IY)
      MINY = MIN0(MINY,IY)
C
C     Insert a  line break  periodically to  prevent  excessively
C     long lines.
C
C     IF (MOD(NPATH,10) .EQ. 0) CALL TKIOWH (2H$N,2)
C
   40 RETURN
      END
