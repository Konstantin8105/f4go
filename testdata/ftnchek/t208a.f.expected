SUBROUTINE          	SUBROUTINE
IDENT               	DDIAPA
(                   	(
IDENT               	IX
,                   	,
IDENT               	IY
)                   	)
NEW_LINE            	

COMMENT             	C     (Add to Path)
NEW_LINE            	

COMMENT             	C     Add the point (IX,IY) to the current path, which is assumed
NEW_LINE            	

COMMENT             	C     to have  already been  started by  a call  to DDIBPA.   The
NEW_LINE            	

COMMENT             	C     current point and visibility are updated in COMMON.
NEW_LINE            	

COMMENT             	C     (27-May-1991)
NEW_LINE            	

COMMENT             	C- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C     EXTERNAL REFERENCES (FUNCTION,SUBROUTINE,COMMON)
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C     EXTERNAL REFS       TKIOLL,      TKIOWB,      TKIOWH,      TKIOWN
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

INTEGER             	INTEGER
IDENT               	TKIOLL
NEW_LINE            	

COMMENT             	C- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C     NON-COMMON VARIABLES
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C     SETS                BLANK
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

INTEGER             	INTEGER
IDENT               	BLANK
,                   	,
IDENT               	DUMMY
,                   	,
IDENT               	IX
,                   	,
IDENT               	IY
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C-----------------------------------------------------------------------
NEW_LINE            	

COMMENT             	C                         P o s t S c r i p t
NEW_LINE            	

COMMENT             	C           D i s p l a y   D e v i c e   I n t e r f a c e
NEW_LINE            	

COMMENT             	C                       C O M M O N   B l o c k
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C     DSSIZE                   display surface size in cm
NEW_LINE            	

COMMENT             	C     IMAGE                    image transformation in effect
NEW_LINE            	

COMMENT             	C     (LASTX,LASTY,VILAST)     last position and visibility of pen
NEW_LINE            	

COMMENT             	C     LINEIN                   line intensity (0..1 scale)
NEW_LINE            	

COMMENT             	C     LINEWT                   line weight (1..25 scale)
NEW_LINE            	

COMMENT             	C     MAGFAC                   resolution magnification factor
NEW_LINE            	

COMMENT             	C     (MOVEX,MOVEY)            position of first point in path
NEW_LINE            	

COMMENT             	C     NPATH                    number of line segments in current page
NEW_LINE            	

COMMENT             	C     PLSTEP                   plotter step size in cm
NEW_LINE            	

COMMENT             	C     (PSXOFF,PSYOFF)          image offset in PostScript units (bp)
NEW_LINE            	

COMMENT             	C     ROTATE                   plot frame is rotated
NEW_LINE            	

COMMENT             	C     (SX,SY)                  scale factors (unit square to plot steps)
NEW_LINE            	

COMMENT             	C     TIMAGE(*,*)              image transformation matrix
NEW_LINE            	

COMMENT             	C     (XMAX,YMAX,ZMAX)         normalized device space extents
NEW_LINE            	

COMMENT             	C     (MAXX,MAXY,MINX,MINY)    actual coordinate limits reached
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C-----------------------------------------------------------------------
NEW_LINE            	

INTEGER             	INTEGER
IDENT               	LASTX
,                   	,
IDENT               	LASTY
,                   	,
IDENT               	LINEWT
,                   	,
IDENT               	MAGFAC
NEW_LINE            	

INTEGER             	INTEGER
IDENT               	MAXX
,                   	,
IDENT               	MAXY
,                   	,
IDENT               	MINX
,                   	,
IDENT               	MINY
NEW_LINE            	

INTEGER             	INTEGER
IDENT               	MOVEX
,                   	,
IDENT               	MOVEY
,                   	,
IDENT               	NPATH
,                   	,
IDENT               	PSXOFF
NEW_LINE            	

INTEGER             	INTEGER
IDENT               	PSYOFF
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

IDENT               	LOGICAL
IDENT               	DVINIT
,                   	,
IDENT               	IMAGE
,                   	,
IDENT               	ROTATE
,                   	,
IDENT               	VILAST
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

IDENT               	REAL
IDENT               	DSSIZE
,                   	,
IDENT               	LINEIN
,                   	,
IDENT               	PLSTEP
,                   	,
IDENT               	SX
NEW_LINE            	

IDENT               	REAL
IDENT               	SY
,                   	,
IDENT               	TIMAGE
,                   	,
IDENT               	XMAX
,                   	,
IDENT               	YMAX
NEW_LINE            	

IDENT               	REAL
IDENT               	ZMAX
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

IDENT               	COMMON
/                   	/
IDENT               	DDI01
/                   	/
IDENT               	DSSIZE
,                   	,
IDENT               	LINEIN
,                   	,
IDENT               	PLSTEP
,                   	,
IDENT               	SX
NEW_LINE            	

IDENT               	COMMON
/                   	/
IDENT               	DDI01
/                   	/
IDENT               	SY
,                   	,
IDENT               	TIMAGE
(                   	(
INT                 	4
,                   	,
INT                 	4
)                   	)
,                   	,
IDENT               	XMAX
,                   	,
IDENT               	YMAX
NEW_LINE            	

IDENT               	COMMON
/                   	/
IDENT               	DDI01
/                   	/
IDENT               	ZMAX
,                   	,
IDENT               	LASTX
,                   	,
IDENT               	LASTY
,                   	,
IDENT               	LINEWT
NEW_LINE            	

IDENT               	COMMON
/                   	/
IDENT               	DDI01
/                   	/
IDENT               	MAGFAC
,                   	,
IDENT               	MAXX
,                   	,
IDENT               	MAXY
,                   	,
IDENT               	MINX
NEW_LINE            	

IDENT               	COMMON
/                   	/
IDENT               	DDI01
/                   	/
IDENT               	MINY
,                   	,
IDENT               	MOVEX
,                   	,
IDENT               	MOVEY
,                   	,
IDENT               	NPATH
NEW_LINE            	

IDENT               	COMMON
/                   	/
IDENT               	DDI01
/                   	/
IDENT               	PSXOFF
,                   	,
IDENT               	PSYOFF
,                   	,
IDENT               	DVINIT
,                   	,
IDENT               	IMAGE
NEW_LINE            	

IDENT               	COMMON
/                   	/
IDENT               	DDI01
/                   	/
IDENT               	ROTATE
,                   	,
IDENT               	VILAST
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

IDENT               	DATA
IDENT               	BLANK
/                   	/
INT                 	32
/                   	/
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C     We can generate an  entry of size "-ddddd -ddddd R "; try to
NEW_LINE            	

COMMENT             	C     keep line lengths under 80 characters.
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

if                  	IF
(                   	(
IDENT               	TKIOLL
(                   	(
IDENT               	DUMMY
)                   	)
>                   	.GT.
INT                 	64
)                   	)
CALL                	CALL
IDENT               	TKIOWH
(                   	(
INT                 	2
IDENT               	H
IDENT               	N
,                   	,
INT                 	2
)                   	)
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C     Use relative coordinates for lines to reduce the  number of
NEW_LINE            	

COMMENT             	C     bytes in the  verbose  PostScript  command language.   This
NEW_LINE            	

COMMENT             	C     shortens the plot file by 5 to 10 percent.
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C     We optimize output into 3 different forms:
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C     # # R   (relative lineto)
NEW_LINE            	

COMMENT             	C     # X     (relative lineto with delta-y = 0)
NEW_LINE            	

COMMENT             	C     # Y     (relative lineto with delta-x = 0)
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C     The  latter two cases  occur frequently enough  to be worth
NEW_LINE            	

COMMENT             	C     taking advantage of  to compress the  output and reduce the
NEW_LINE            	

COMMENT             	C     parsing  time.    We cannot  discard zero length  segments,
NEW_LINE            	

COMMENT             	C     however, because that  would discard single  points from  a
NEW_LINE            	

COMMENT             	C     point plot.
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

if                  	IF
(                   	(
IDENT               	IX
==                  	.EQ.
IDENT               	LASTX
)                   	)
IDENT               	GO
IDENT               	TO
INT                 	10
NEW_LINE            	

if                  	IF
(                   	(
IDENT               	IY
==                  	.EQ.
IDENT               	LASTY
)                   	)
IDENT               	GO
IDENT               	TO
INT                 	20
NEW_LINE            	

CALL                	CALL
IDENT               	TKIOWN
(                   	(
IDENT               	IX
-                   	-
IDENT               	LASTX
)                   	)
NEW_LINE            	

CALL                	CALL
IDENT               	TKIOWB
(                   	(
IDENT               	BLANK
)                   	)
NEW_LINE            	

CALL                	CALL
IDENT               	TKIOWN
(                   	(
IDENT               	IY
-                   	-
IDENT               	LASTY
)                   	)
NEW_LINE            	

CALL                	CALL
IDENT               	TKIOWH
(                   	(
INT                 	3
IDENT               	H
IDENT               	R
,                   	,
INT                 	3
)                   	)
NEW_LINE            	

IDENT               	GO
IDENT               	TO
INT                 	30
NEW_LINE            	

INT                 	10
CALL                	CALL
IDENT               	TKIOWN
(                   	(
IDENT               	IY
-                   	-
IDENT               	LASTY
)                   	)
NEW_LINE            	

CALL                	CALL
IDENT               	TKIOWH
(                   	(
INT                 	3
IDENT               	H
IDENT               	Y
,                   	,
INT                 	3
)                   	)
NEW_LINE            	

IDENT               	GO
IDENT               	TO
INT                 	30
NEW_LINE            	

INT                 	20
CALL                	CALL
IDENT               	TKIOWN
(                   	(
IDENT               	IX
-                   	-
IDENT               	LASTX
)                   	)
NEW_LINE            	

CALL                	CALL
IDENT               	TKIOWH
(                   	(
INT                 	3
IDENT               	H
IDENT               	X
,                   	,
INT                 	3
)                   	)
NEW_LINE            	

INT                 	30
IDENT               	NPATH
=                   	=
IDENT               	NPATH
+                   	+
INT                 	1
NEW_LINE            	

IDENT               	LASTX
=                   	=
IDENT               	IX
NEW_LINE            	

IDENT               	LASTY
=                   	=
IDENT               	IY
NEW_LINE            	

IDENT               	VILAST
=                   	=
IDENT               	.TRUE.
NEW_LINE            	

IDENT               	MAXX
=                   	=
IDENT               	MAX0
(                   	(
IDENT               	MAXX
,                   	,
IDENT               	IX
)                   	)
NEW_LINE            	

IDENT               	MINX
=                   	=
IDENT               	MIN0
(                   	(
IDENT               	MINX
,                   	,
IDENT               	IX
)                   	)
NEW_LINE            	

IDENT               	MAXY
=                   	=
IDENT               	MAX0
(                   	(
IDENT               	MAXY
,                   	,
IDENT               	IY
)                   	)
NEW_LINE            	

IDENT               	MINY
=                   	=
IDENT               	MIN0
(                   	(
IDENT               	MINY
,                   	,
IDENT               	IY
)                   	)
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C     Insert a  line break  periodically to  prevent  excessively
NEW_LINE            	

COMMENT             	C     long lines.
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

COMMENT             	C     IF (MOD(NPATH,10) .EQ. 0) CALL TKIOWH (2H$N,2)
NEW_LINE            	

COMMENT             	C
NEW_LINE            	

INT                 	40
return              	RETURN
NEW_LINE            	

END                 	END
NEW_LINE            	

